File: Problem-23/InsertElement.java
// WAP to enter an element at specific position into array. (Do not take a new array)

import java.util.Scanner;

public class InsertElement{
    public static void main(String[] args){

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter n : ");
        int n = sc.nextInt();

        int[] arr = new int[n + 1];

        System.out.println("Enter elements into arr : ");
        for(int i = 0; i < n; i++){
            System.out.println("arr["+i+"] : ");
            arr[i] = sc.nextInt();
        }

        System.out.print("Enter the element : ");
        int element = sc.nextInt();

        System.out.print("Enter the position : ");
        int position = sc.nextInt();

        if(position < 1 || position > n + 1){
            System.out.println("Oops...can't insert an element...Plss Enter valid position !!");
        }

        else{  
            for(int i = n; i >= position; i--){
                arr[i] = arr[i - 1];
            }
    
            arr[position - 1] = element;
            n++;

            System.out.println("Array after insertion :");

            for(int i = 0; i < n; i++){
                System.out.print(arr[i] + " ");
            }
        }
    }
}

File: Problem-23/InsertElement.py
# WAP to enter an element at specific position into array. (Do not take a new array)

n = int(input("Enter n : "))
arr = []

print("Enter elements into arr : ")
for i in range(n) : 
    arr.append(int(input(f"arr[{i}] : ")))

element = int(input("Enter the element : "))
position = int(input("Enter the position : "))

if( position < 1 or position > n + 1) :
    print("Oops...can't insert an element...Plss Enter valid position !!")

else :
    arr.append(0)

    for i in range(n, position - 1, -1) :
        arr[i] = arr[i - 1]

    arr[position - 1] = element
    
print("Array after insertion : ")
for i in arr :
    print(i, end="  ")